x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityRestIT.java

```diff
@@ -327,6 +344,25 @@ public void testNodesInfo() throws IOException {
         });
     }
 
+    public void testRestApiKeyIsNotSupportedForCrossClusterAccess() throws IOException {
+        updateClusterSettings(
+            randomBoolean()
+                ? Settings.builder()
+                    .put("cluster.remote.wrong_api_key_type.seeds", fulfillingCluster.getRemoteClusterServerEndpoint(0))
+                    .build()
+                : Settings.builder()
+                    .put("cluster.remote.wrong_api_key_type.mode", "proxy")
+                    .put("cluster.remote.wrong_api_key_type.proxy_address", fulfillingCluster.getRemoteClusterServerEndpoint(0))
+                    .build()
+        );
+        final ResponseException e = expectThrows(
+            ResponseException.class,
+            () -> performRequestWithRemoteSearchUser(new Request("GET", "/wrong_api_key_type:*/_search"))
+        );
+        assertThat(e.getResponse().getStatusLine().getStatusCode(), equalTo(401));
+        assertThat(e.getMessage(), containsString("unable to authenticate user "));
```

Comment:
Not for this PR but currently, and really more a musing than an actionable suggestion. 

Currently, if a REST API key is configured (in place of a cross-cluster one) this is actually not that easy to detect, for a QC-side admin. 

All they get are generic 401 messages on CCS (and other remote requests), without the extra context that this is due to the wrong API key type (I think).

This isn't something we introduced with this change (same problem exists for e.g., key not found) and may be kind of tricky to address. So perhaps the best way to deal with this is just having a trouble-shooting section in docs that emphasizes common causes for a 401 here... 

Just jotting this down. As I said, not something to worry about now.

Reply:
Errr this turns out to be a test bug. Thanks for cathing it!

With #94998, we should report underlying error in the response. The real issue here is that the `remote_search` user was not created on the querying cluster. It was a native user created in the beginning of `testCrossClusterSearch` method and I didn't copy it in the new method. So It was actually reporting 401 of the `remote_search` on the querying cluster. I fixed it by merging it back to `testCrossClusterSearch`. Now the error response contains message about mismatched API key type. Also these errors are logging on the QC side so that QC admins will be able to see them right after configuring the remote cluster because the handshake would fail for the same error. In summary, I think the error should be pretty visible on the QC side.